### PR TITLE
Adds the Cat and Dog Person traits

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: You can now be a cat person or a dog person in Space Station 13!
/:cl:

Thaw PR: #36663

You know how you ask someone their favorite kind of pets, and they might say something like "oh, I'm a cat person?" Now you can be one. These two new neutral traits give you preference to dogs or cats, meaning you gain a slightly higher mood increase from petting your preferred one and slightly lower mood increase from petting the ~~inferior~~ other one. So, if you're a dog person, you get +5 to mood from petting a corgi and +1 from petting a cat, instead of +3 from either like normal.

Having both traits will cancel each other out. Also changed the `pet_corgi` association for moodlets gained by petting animals to `wuv` since there's now a mood increase from petting cats, too.

### CODE REVIEW

**Comment by JJRcop** [comment]
This should have some interaction with cat mutant parts.

**Comment by KorPhaeron** [comment]
You need to fix moodlet traits being available when moodlets are not before you add more of them

**Comment by Anonmare** [comment]
Lucky for you I'm a dog lover

**Comment by Xhuis** [comment]
@KorPhaeron I've been intending to make an "is_available" system for traits, but I'm not sure how to implement them since in its current state the objects aren't created, just read from their initial vars. Would it be cleaner to use those initial vars like `incompatible_with` and `moodlet_trait`, or to have a separate datum for the purchase entry that has an `is_available()` proc that can be overriden?

**Comment by nicbn** [comment]
In other words free mood by petting mobs

**Comment by Xhuis** [comment]
Mood boosts from petting corgis has been around since moodlets were first added.

**Comment by ma44** [comment]
What if you were a dog lover but a cat person.

**Comment by Xhuis** [comment]
Having both traits cancels each other out and makes no change to either mood bonus from its normal value.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Xhuis replace mood with braindamage when mood is disabled, problem solved :^)

**Comment by Xhuis** [comment]
I made mood traits be unavailable for purchase if moodlets are disabled, plus a failsafe that removes them from your setup if they're disabled but you added them before they were turned off.

Also some more tweaks to Social Anxiety, the eternal conundrum.

**Comment by ExcessiveUseOfCobblestone** [comment]
why would you tie the fix to a feature pr you hecker

**Comment by Xhuis** [comment]
I can only make one thaw PR during the freeze and the bundle of features here is what I'd wanted to add once it ended. Unless any maintainers have a notable problem with it and want me to slice off the Social Anxiety portion, the other two are fine in the same PR, I think.

**Comment by Xhuis** [comment]
I'll stab my light knife through your heart

**Comment by ExcessiveUseOfCobblestone** [comment]
watch it buddy i'll take your heirloom.

**Comment by iskyp** [comment]
image
i read this and thought moodlets were disabled for a second because there isn't an if between "traits" and "moodlets"

**Comment by Xhuis** [comment]
Oops.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
Please atomize this pr

**Comment by Xhuis** [comment]
As I already mentioned, the feature freeze prevents me from atomizing all of it. If I wanted to add the Cat and Dog Person traits in another PR, I'd have to add the framework separately in a different PR since the thaw only covers one PR.

I can remove the social anxiety tweaks if that's your main concern, though. Otherwise, I'm not sure the best way to handle it.

**Comment by ninjanomnom** [comment]
I primarily meant splitting up the code disabling traits, the anxiety changes, and the [animal] person traits. You can include the framework for whatever changes you need in those.

**Comment by Xhuis** [comment]
There we go, quick and clean.

**Comment by ExcessiveUseOfCobblestone** [comment]
You've already secured the spot in the thaw, make the fix first.

**Comment by ninjanomnom** [comment]
Kor requested the traits get their fix before more features are merged so this'll have to wait until that gets merged.

Or at least approved

**Comment by Xhuis** [comment]
I'll make a second PR that includes the framework/fix standalone, and once that gets in I can update this one.

File: code/modules/mob/living/simple_animal/friendly/cat.dm
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: uhm this is all backwards.

Send only a signal that a cat has been petted.

then let the thing that processes the signal figure out what trait the person has.

You're leaking the systems abstractions

**Review by optimumtact — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/cat.dm
Comment (Xhuis) [reply code/modules/mob/living/simple_animal/friendly/cat.dm]: Unless I misunderstand moodlet code, wouldn't that require me to check every moodlet event to see if it's a certain type, and if so, if the traits were present?

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/friendly/cat.dm]: Currently, yes. You will have to edit moodlet code.

Comment (optimumtact) [reply code/modules/mob/living/simple_animal/friendly/cat.dm]: @Qustinnus didn't I make you do this for something else, how did you implement it?

**Comment by Xhuis** [comment]
I don't have any more desire to work on this or wait for more days hoping Qustinnus will respond so I'm gonna close this

**Comment by Qustinnus** [comment]
Oh im sorry I havnt been on github for a while due to personal issues and uni keeping me busy. Ill look into it now

File: code/modules/mob/living/simple_animal/friendly/cat.dm
Comment (Qustinnus) [reply code/modules/mob/living/simple_animal/friendly/cat.dm]: I think my case was unique because everyone gets the same moodlet (table) except I modified it after if you are a catperson. Ill look into a different way of approaching this

**Pull request closed without merging**
